packages/dds/merge-tree/Obliterate.md

```diff
@@ -0,0 +1,276 @@
+  will not result in that inserted segment being deleted or annotated. For example, from an initial state "012",
+  the operations [delete the range [1, 3)] from client 1 and [insert "hi" at index 2 (i.e. between "1" and "2")] from client 2,
+  the resulting text is "0hi".
+
+The merge outcomes for ranges are easy to understand, but not always desirable.
+Oftentimes, when consumers want to work with ranges, they may want their operation to apply to concurrently inserted segments.
+One set of implemetable semantics would be rather than the range operation applying to a range of character positions at the
+time the operation is issued, it applies at the time the operation is sequenced.
+Consider that:
+
+```
+// Initial state at seq 0: "012"
+{ seq: 1, refSeq: 0, clientId: 1, op: <insert "hi" at index 2> }
+{ seq: 2, refSeq: 0, clientId: 2, op: <delete the range [1, 3)> }
+```
+
+would result in the text "0". However, if those concurrent operations were sequenced in the opposite order:
+
+```
+// Initial state at seq 0: "012"
+{ seq: 1, refSeq: 0, clientId: 2, op: <delete the range [1, 3)> }
+{ seq: 2, refSeq: 0, clientId: 1, op: <insert "hi" at index 2> }
+```
+
+we'd still end up with the text "0hi".
+
+One option is even more extreme: not only does the range operation apply to the range at the time the op is sequenced,
+it also applies to any subsequent segments that get concurrently inserted into this range.
+Under these semantics, both orders of sequencing the above operations would result in the text "0".
+
+This final set of merge semantics is known as "obliterate".
+
+Notice that the above examples always insert text at positions strictly inside the removed range.
+If the insert operation was instead before the "1" or after the "2", one can imagine different applications wanting different behavior:
+either the obliterated region should expand to include that text, or it should not.
+This topic will be covered in the [endpoint behavior](#endpoint-behavior) section,
+but for implementation strategy discussion one should assume that the implementation should support both options (and leave it up to applications
+to specify).
+
+## Implementation Strategy
+
+This section is focused on implementation of the "obliterate" semantics inside merge tree.
+This will constitute the bulk of the complexity of the feature.
+Since obliterate is generally a "different kind of remove," there may be a nice abstraction to introduce at the code level to generalize
+removal information. However, in favor of introducing niceties later this design document will assume fields are inlined and focus on
+the strategy for ensuring eventual consistency. If such an abstraction is introduced, ideally it would enable better "pay-to-play" of
+common code paths based on merge-tree feature usage.
+As an example, `BaseSegment.split()` needs to copy segment properties to the split segment.
```

Comment:
something like #81 would help here. wondering if it is worth doing that work as a pre-requisite for obliterate